lib/db/leveldb.go

```diff
@@ -408,6 +427,12 @@ func ldbUpdateGlobal(db dbReader, batch dbWriter, folder, device []byte, file pr
 			panic(err)
 		}
 
+		if len(fl.versions) > 0 {
+			// Keep the current neweset file around so we can subtract it from
+			// the globalSize if we replace it.
+			oldFile, hasOldFile = ldbGet(db, folder, fl.versions[0].device, name)
```

Comment:
This get could go into the `if insertedAt == 0 {` branch, so that we don't do a get if we haven't inserted at the top.
Though inserting has to be deferred after that, or we have to be able to get a file at index 1.